mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxQuery.java

```diff
@@ -136,20 +128,43 @@ private MailboxPath buildBase() {
      * @param pathDelimiter
      *            path delimiter to use
      */
-    @VisibleForTesting MailboxQuery(MailboxPath base, String expression, char pathDelimiter) {
-        super();
-        this.base = base;
-        if (base.getName() == null)
-            this.base.setName("");
+    @VisibleForTesting MailboxQuery(Optional<String> namespace, Optional<String> user, Optional<String> name,
+                                    String expression, MailboxSession session) {
+        this.namespace = namespace;
+        this.user = user;
+        this.name = name;
+        this.base = new MailboxPath(
+            namespace.orElse(MailboxConstants.USER_NAMESPACE),
+            user.orElse(session.getUser().getUserName()),
+            name.orElse(EMPTY_PATH_NAME));
         if (expression == null) {
             this.expression = "";
         } else {
             this.expression = expression;
         }
-        this.pathDelimiter = pathDelimiter;
+        this.pathDelimiter = session.getPathDelimiter();
         pattern = constructEscapedRegex();
     }
 
+    public MailboxPath getPathLike() {
+        String combinedName = getCombinedName()
+            .replace(getFreeWildcard(), SQL_WILDCARD_CHAR)
```

Comment:
Why with the current value of your constant this line is equivalant to:
```.replace(getFreeWildcard(), LOCAL_WILDCARD)``
either there is something I really do not understand either it's a big mistake

Reply:
Ok I think I understand why, this is because when you do:
```
mailboxMapper
+            .findMailboxWithPathLike(mailboxExpression.getPathLike())
```
in ``findMailboxWithPathLike`` "%" have the same meaning that "*" in this class. 
And for implementing "%" of this class, you start by requesting more at the mailboxMapper level and than do a filter on memory here.

This is hard to understand, what I will do to make it more understandable is moving the SQL_WILDCARD_CHAR constant into MailboxMapper and rename it FREE_WILDCARD.
Also we could even considere changing it's value from '%' to '*' but I guess it will breaks some API

Reply:
this is not related to this pr, and anyway we don't want to fix it for now